test/test_jit.py

```diff
@@ -4279,6 +4279,57 @@ def test_index_put(target, indices, rhs):
 
         self.assertExpected(str(test_index_put.graph))
 
+    @unittest.skip('https://github.com/pytorch/pytorch/issues/9595')
+    def test_annotated_script_fn(self):
+        @torch.jit.script
+        def foo(x, y, z):
+            # type: (Tensor, Tuple[Tensor, Tensor, Tensor], Tuple[Tensor, Tuple[Tensor, Tensor]]) -> Tensor
+            return x
+
+        self.assertExpected(foo.__getattr__('forward').pretty_print_schema())
+
+    @unittest.skip('https://github.com/pytorch/pytorch/issues/9595')
+    def test_annotated_script_method(self):
+        class SM(torch.jit.ScriptModule):
+            @torch.jit.script_method
+            def forward(self, x, y):
+                # type: (Tuple[Tensor, Tensor], Tensor) -> Tuple[Tensor, Tensor, Tensor]
+                return y, y, y
+
+        sm = SM()
+
+        self.assertExpected(sm.__getattr__('forward').pretty_print_schema())
```

Comment:
I see how running those functions is not supported yet, but do we seriously can't even compile them at the moment? That's very weird, because we have hardly any tests for this patch.

Reply:
@zdevito and I discussed offline. Extending support for inlining + executing ScriptModules with non-`Tensor` inputs is going to involve some refactoring and it's probably better to split up the work rather than cram that (and a bunch of other changes) into this PR

Reply:
I agree, and that makes sense. However, this function is only trying to compile them without actually running them, so why are those tests skipped?

Reply:
The checks in `LowerTuples` that ensure that graph inputs and outputs are not tuples fails. How about I move that check to `GraphExecutor` construction so we can still test that compilation works?